tests/framework/extensions/machinepools/machinepools.go

```diff
@@ -192,3 +197,95 @@ func ScaleMachinePoolNodes(client *rancher.Client, cluster *v1.SteveAPIObject, n
 
 	return scaledClusterResp, nil
 }
+
+// AddRKE2K3SCustomClusterNodes is a helper method that will add nodes to the custom RKE2/K3S custom cluster.
+func AddRKE2K3SCustomClusterNodes(client *rancher.Client, cluster *v1.SteveAPIObject, nodes []*nodes.Node, rolesPerNode []string) error {
+	clusterStatus := &apiv1.ClusterStatus{}
+	err := v1.ConvertToK8sType(cluster.Status, clusterStatus)
+	if err != nil {
+		return err
+	}
+
+	token, err := tokenregistration.GetRegistrationToken(client, clusterStatus.ClusterName)
+	if err != nil {
+		return err
+	}
+
+	var command string
+	for key, node := range nodes {
+		logrus.Infof("Adding node %s to cluster %s", node.NodeID, cluster.Name)
+		if strings.Contains(rolesPerNode[key], "windows") {
+			command = fmt.Sprintf("powershell.exe %s -Address %s", token.InsecureWindowsNodeCommand, node.PublicIPAddress)
+		} else {
+			command = fmt.Sprintf("%s %s --address %s", token.InsecureNodeCommand, rolesPerNode[key], node.PublicIPAddress)
+		}
+
+		output, err := node.ExecuteCommand(command)
+		if err != nil {
+			return err
+		}
+
+		logrus.Infof(output)
+	}
+
+	err = kwait.Poll(500*time.Millisecond, defaults.TenMinuteTimeout, func() (done bool, err error) {
+		clusterResp, err := client.Steve.SteveType("provisioning.cattle.io.cluster").ByID(cluster.ID)
+		if err != nil {
+			return false, err
+		}
+
+		if clusterResp.ObjectMeta.State.Name == active && nodestat.AllManagementNodeReady(client, cluster.ID, defaults.ThirtyMinuteTimeout) == nil {
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// DeleteRKE2K3SCustomClusterNodes is a helper method that will delete nodes from the custom RKE2/K3S custom cluster.
+func DeleteRKE2K3SCustomClusterNodes(client *rancher.Client, clusterID string, cluster *v1.SteveAPIObject, nodesToDelete []*nodes.Node) error {
+	steveclient, err := client.Steve.ProxyDownstream(clusterID)
+	if err != nil {
+		return err
+	}
+
+	nodesSteveObjList, err := steveclient.SteveType("node").List(nil)
+	if err != nil {
+		return err
+	}
+
+	for _, nodeToDelete := range nodesToDelete {
+		for _, node := range nodesSteveObjList.Data {
+			if node.Annotations[externalIP] == nodeToDelete.PrivateIPAddress {
+				machine, err := client.Steve.SteveType("cluster.x-k8s.io.machine").ByID(namespace + "/" + node.Annotations["cluster.x-k8s.io/machine"])
```

Comment:
there should be a const for these

Reply:
They exist in clusters.go, but it causes an import cycle issue in Go. We had this problem in a previous PR, so we just did it like this.

We can always just make a new const, but it may cause confusion later.